operator/main.go

```diff
@@ -98,10 +98,10 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
-	setupManager(flagVar, controllers.NewCacheFunc(), scheme)
+	setupManager(flagVar, controllers.NewCacheFunc(), scheme, &opts)
 }
 
-func setupManager(flagVar *FlagVar, newCacheFunc cache.NewCacheFunc, scheme *runtime.Scheme) {
+func setupManager(flagVar *FlagVar, newCacheFunc cache.NewCacheFunc, scheme *runtime.Scheme, opts *zap.Options) {
```

Comment:
instead of passing zap, I think we should pass logr.Logger since it is used everywhere else. also you can simply use ctrl.LoggerFrom in your code and it will use the singleton from the set logger. That way you dont have to pass it around in the code on every parameter :)

Reply:
Thanks for the tip! I will refactor it after this [PR](https://github.com/kyma-project/kyma-watcher/pull/12) gets merged. :)